infor-design/enterprise-wc
# 899 - Datagrid Xss, Nested Data and Formatters and Virtual Scroll

**Explain the details for making this change. What existing problem does the pull request solve?**
This PR adds a few general features needed as a base and starting point for #737. The following features and fixes are added.

- Ability to supported nested data in the dataset `data.child.setting` for example
- Full test covered (enabled paging tests)
- Xss prevention
- Checkbox Formatter
- Custom Formatter
- Fixes to Virtualization to support selection and selection indexes
- Refactor htmlEntities to escapteHTML (double functions existed)

**Related github/jira issue (required)**:
Fixes #899 
Fixes #862 
Fixes #627
Some of #737 

**Steps necessary to review your pull request (required)**:
- go to http://localhost:4300/ids-data-grid/data-nested.html and see that all columns are rendering a value even on tho some data is nested 2 and 3 levels deep in the data array
- go to http://localhost:4300/ids-data-grid/xss-prevention.html and check the console for no errors (should be no csp errors). This page will now render with "xss" "attacks" in it escaping the data
- tests and checks all pass (on the ci)
- go to http://localhost:4300/ids-data-grid/columns-formatters.html and observe the new checkbox formatter and the custom formatter (last one). The custom formatter shows how to roll your own formatter.
- go to http://localhost:4300/ids-data-grid/virtual-scroll.html
- select a row near the top and bottom
- check the console that the row's selectedRows index is correct
- scroll back up and see the selection is maintained
- try and sort the columns

**Included in this Pull Request**:
